activerecord/lib/active_record/database_configurations/hash_config.rb

```diff
@@ -86,7 +89,7 @@ def min_threads
       end
 
       def max_threads
-        (configuration_hash[:max_threads] || max_connections).to_i
+        (configuration_hash[:max_threads] || max_connections || 5).to_i
```

Comment:
@byroot any thoughts on this?

Reply:
I'm tempted to suggest:

```suggestion
        (configuration_hash[:max_threads] || 5).to_i
```

As you probably want to ensure the async query threads can't starve the web server threads out of connections. So seeing it now I feel like this default wasn't great.

Reply:
I found a way to overcomplicate it instead 😅 

https://github.com/rails/rails/blob/fed08be767b2494950c1c2b5489f77136504221a/activerecord/lib/active_record/database_configurations/hash_config.rb#L92

wdyt?

Reply:
Sure.